### ISSUE TITLE
Question about “locals exceed maximum”

### ISSUE THREAD

**Issue by SDGGiesbrecht** [swiftwasm/carton#208]
`carton test` fails with the following:

```
error: failed to run `[...]/.build/wasm32-unknown-wasi/debug/[Package]Tests.wasm`
│   1: module instantiation failed (engine: jit, compiler: cranelift)
╰─> 2: Validation error: locals exceed maximum (at offset 1442120)
```

I don’t understand what that means or how to go about finding the underlying problem.

**Comment by MaxDesiatov** [comment]
Thanks! This is one of the errors we've been chasing for some time, but couldn't find an isolated reproducible test case. Do you have one to share here by any chance? Either an isolated file compilable with `swiftc` or a SwiftPM project would be much appreciated!

**Comment by SDGGiesbrecht** [comment]
Yes, I managed to narrow the problem down to a particular long dictionary literal. Let me clean‐up some of the mess left over from my experiments, and then I’ll post a link.

**Comment by MaxDesiatov** [comment]
I don't think it's related to `carton` directly, so I'm moving it to the toolchain repo in the meantime...
